### PR TITLE
Edit Nginx config

### PR DESCRIPTION
**Pull request by LAV45**
disable php engine for the folder /assets

### CODE REVIEW

File: docs/guide/start-installation.md [hunk 1]
```diff
@@ -156,16 +161,21 @@ the installed application. You only need to do these once for all.
            #    try_files $uri =404;
            #}
            #error_page 404 /404.html;
-       
+
+           # disable php engine for the folder /assets
+           location ~ ^/assets {
+              fastcgi_param PHP_FLAG "engine=on";
```
Comment (cebe) [inline docs/guide/start-installation.md]: why disable php engine? it would be better to just not pass it to fcgi at all.

**Comment by LAV45** [comment]
If you call the file ```/assets/6de2e167/js/back_door.php``` it falls under the rule ```location ~ \.php$ {}``` will be executed

**Comment by cebe** [comment]
Check the nginx docs:

> http://nginx.org/en/docs/http/request_processing.html
>
> nginx first searches for the most specific prefix location given by literal strings regardless of the listed order. In the configuration above the only prefix location is “/” and since it matches any request it will be used as a last resort. Then nginx checks locations given by regular expression in the order listed in the configuration file. **The first matching expression stops the search and nginx will use this location.** If no regular expression matches a request, then nginx uses the most specific prefix location found earlier. 

since you are using a regex expression instead of a prefix you can just avoid passing it to CGI here.

**Comment by LAV45** [comment]
@cebe , what do you think about this variant?

**Comment by samdark** [comment]
@LAV45 I don't like this change. Could be done simpler by defining location for static files and never passing it to fastcgi:

```
location ~* \.(jp?g|jpe|gif|png|txt|swf|xml|vnd|djv?|tif?|ico|bmp|mp3|rar|zip|avi|wmv|mpg|7z|mov|mp4|arj|js|css|html|chm|wsz)$ {
        root /var/www/yiiframework/frontend/web;
        expires max;
        break;
    }
```

**Comment by LAV45** [comment]
@samdark you offer to allow all valid files from the assets folder while the risk is only one type of php files. I think your option is highly redundant.

**Comment by samdark** [comment]
1. Valid asset files are usually of the same types so the list could be reused.
2. It's simpler than your proposal. It breaks early.
3. Whitelisting is more secure in general.
4. You can adjust that regexp to match everything that's not `.php`.

**Comment by LAV45** [comment]
:+1:

**Pull request closed without merging**